pyccel/codegen/printing/fcode.py

```diff
@@ -2697,12 +2790,40 @@ def split_pos_code(line, endpos):
         result = []
         trailing = ' &'
         for line in lines:
-            if len(line)>72 and ('"' in line[72:] or "'" in line[72:] or '!' in line[:72]):
+            if len(line)>72 and '!' in line[:72]:
                 result.append(line)
-
+            elif len(line)>72 and ('"' in line[72:] or "'" in line[72:]):
+                pos=[]
+                quote = 0
+                double_quote = 0
+                mine = []
+                for idx, c in enumerate(line):
+                    if c == '\'':
+                        quote += 1
+                    if c == '\"':
+                        double_quote += 1
+                    if c == ' ' and quote%2 == 0 and double_quote %2 == 0 and idx > 72 and (not pos or idx > pos[-1] + 65):
+                        pos.append(idx)
```

Comment:
I wonder if some of this can be accomplished with [regexes](https://docs.python.org/3/library/re.html)

Reply:
Sorry, i just wanted to make sure the tests worked before cleaning up the code. I will refactor it using regex

Reply:
No problem, do it in whatever order works best for you, I just thought it was worth mentioning regexes sooner rather than later because they can be a pain to use and will probably replace a large chunk of code, so it may not be worth working at fixing this bit if you are going to replace it as soon as it works

Reply:
Looks like it's working now anyway :+1:

Reply:
Finally, i still don't think it's enough. Working on the parity of quote and double quote to know when it is safe to break a line will not account for the use of \' or \'' inside a quote or a double quote. I need to make sure I am only counting the actual quotes.

Reply:
You need to activate the tests for this to make sure you are handling these cases

Reply:
[This file](https://github.com/pyccel/pyccel/blob/master/tests/pyccel/scripts/print_strings.py) and [this file](https://github.com/pyccel/pyccel/blob/master/tests/pyccel/scripts/print_sp_and_end.py) contain lots of test cases that seem relevant to this PR. You need to make sure these are activated [here](https://github.com/pyccel/pyccel/blob/master/tests/pyccel/test_pyccel.py#L612-L623)

You could also add additional tests with single/double quotes and longer lines

Reply:
I think it should be manageable with a regex that looks something like [this](https://stackoverflow.com/a/5696141)

Reply:
I recommend using [raw strings](https://www.askpython.com/python/string/python-raw-strings) for this